### ISSUE TITLE
VSCode "Rust Analyzer: Run" command sometimes loses the "run" command

### ISSUE THREAD

**Issue by kvelicka** [rust-analyzer/rust-analyzer#4373]
Normally, when using `Rust Analyzer: Run` I get presented with three options: run, check and test. However, sometimes the run option simpy disappears from the list and I can only do check and test.

It's of course possible that I'm missing something obvious here - apologies if that is the case.

I'm running rust-analyzer version 0.2.159

**Comment by bnjjj** [comment]
Do you have some use case or pattern to reproduce this bug ?

**Comment by kvelicka** [comment]
I didn't notice anything specific but the problem seems rather persistent - it affects at least two different Rust projects and persists over restart of the VSCode-started language server process.

Is there some way I can get more info/logs useful for debugging? I'm happy to help with digging deeper on this but I'm unsure about how I can do that.

**Comment by matklad** [comment]
Run is only present if the cursor is inside `main` function, it seems like this could be the issue?

**Comment by matklad** [comment]
Though, I think that it maybe fine to just include it anywhere in the `bin` crate?

**Comment by kvelicka** [comment]
> Run is only present if the cursor is inside main function, it seems like this could be the issue?

That is indeed the case, thanks!

> Though, I think that it maybe fine to just include it anywhere in the `bin` crate?

FWIW that's how I expected it to work, but perhaps there's a reason for it to work this way that I'm not aware of.

My issue as stated is resolved but I'll keep it open in case anything is decided re: including it anywhere in `bin`. Feel free to close this if not.

**Comment by matklad** [comment]
Yeah, I feel we should enable it in bins/examples (but we should keep the code lens on the literal `fn main` function)
